src/RipQP.jl

```diff
@@ -37,6 +37,42 @@ include("utils.jl")
 
 const to = TimerOutput()
 
+keep_init_prec(mode::Symbol) = (mode == :mono || mode == :ref || mode == :zoom)
+
+include("mono_solver.jl")
+include("two_steps_solver.jl")
+include("three_steps_solver.jl")
+
+function ripqp(
```

Comment:
@tmigot I made these changes following our discussion (there are a lot of modifications but you can look at this short function only).
`RipQPSolver` uses dispatch based on `ap::AbstractRipQPParameters` (there are different parameters structures used to change the settings to solve the problem).
`stats_inner` are the stats of the preprocessed QuadraticModel `QM_inner`, and `get_stats_outer` returns the stats of the initial QuadraticModel `QM`.

Reply:
In the docstring of `ripqp`, you can also add a block like this (inspired form https://github.com/JuliaSmoothOptimizers/JSOSolvers.jl/blob/main/src/R2.jl#L8 ):
```
For advanced usage, first define a `R2Solver` to preallocate the memory used in the algorithm, and then call `solve!`:
    solver = R2Solver(nlp)
    solve!(solver, nlp; kwargs...)
```